mi/dataset/parser/metbk_a_dcl.py

```diff
@@ -124,6 +81,21 @@ def __init__(self, raw_data, *args, **kwargs):
             INSTRUMENT_PARTICLE_MAP,
             *args, **kwargs)
 
+    def _build_parsed_values(self):
+        """
+        Build parsed values for Recovered and Telemetered Instrument Data Particle.
+        Will only append float values and ignore strings.
+        Returns the list.
+        """
+        data_list = []
+        for name, group, func in INSTRUMENT_PARTICLE_MAP:
+            if isinstance(self.raw_data[group], float):
+                data_list.append(self._encode_value(name, self.raw_data[group], func))
```

Comment:
since func has to be float, recommend making things match:

```python
    for name, group, func in INSTRUMENT_PARTICLE_MAP:
        if isinstance(self.raw_data[group], func):
            data_list.append(self._encode_value(name, self.raw_data[group], func))
```

Reply:
since there is a check to make sure the particle matches the required type, why are we generating the map each time we parse a single line? more efficient to simply create a static list, no?